config/ember-try.js

```diff
@@ -1,72 +1,89 @@
   return {
-    useYarn: true,
-    scenarios: [
-      {
-        name: "ember-lts-3.28",
-        npm: {
-          devDependencies: {
-            "ember-source": "~3.28.1",
-          },
-        },
-      },
-      {
-        name: "ember-lts-4.4",
-        npm: {
-          devDependencies: {
-            "ember-source": "~4.4.0",
-          },
-        },
-      },
-      {
-        name: "ember-release",
-        npm: {
-          devDependencies: {
-            "ember-source": await getChannelURL("release"),
-          },
-        },
-      },
-      {
-        name: "ember-beta",
-        npm: {
-          devDependencies: {
-            "ember-source": await getChannelURL("beta"),
-          },
-        },
+    ...config,
+    scenarios: config.scenarios.map((scenario) => ({
+      ...scenario,
+      name: `ember-lts-${version} (${scenario.name})`,
+      npm: {
+        devDependencies: { "ember-source": `~${version}.0` },
       },
-      {
-        name: "ember-canary",
-        npm: {
-          devDependencies: {
-            "ember-source": await getChannelURL("canary"),
-          },
-        },
-      },
-      {
-        name: "ember-classic",
-        env: {
-          EMBER_OPTIONAL_FEATURES: JSON.stringify({
-            "application-template-wrapper": true,
-            "default-async-observers": false,
-            "template-only-glimmer-components": false,
-          }),
-        },
-        npm: {
-          devDependencies: {
-            "ember-source": "~3.28.0",
-          },
-          ember: {
-            edition: "classic",
-          },
-        },
+    })),
+  };
+}
+
+async function buildURLScenario(release) {
+  const emberSourceURL = await getChannelURL(release);
+
+  return {
+    ...config,
+    scenarios: config.scenarios.map((scenario) => ({
+      ...scenario,
+      name: `ember-${release} (${scenario.name})`,
+      npm: {
+        devDependencies: { "ember-source": emberSourceURL },
       },
-      embroiderSafe(),
-      embroiderOptimized(),
-    ],
+    })),
+  };
+}
+
+function buildEmbroiderScenario(type) {
+  const embroiderScenario = embroiderTestSetup[type]();
+
+  return {
+    ...config,
+    scenarios: config.scenarios.map((scenario) => ({
+      ...scenario,
+      ...embroiderScenario,
+      name: `${embroiderScenario.name} (${scenario.name})`,
+    })),
   };
+}
+
+module.exports = function () {
+  const scenario = process.env.EMBER_SCENARIO;
+
+  if (/^ember-lts-\d+\.\d+/.test(scenario)) {
+    return buildLTSScenario(scenario.replace(/^ember-lts-/, ""));
+  } else if (/^ember-(release|beta|canary)$/.test(scenario)) {
+    return buildURLScenario(scenario.replace(/^ember-/, ""));
+  } else if (/^embroider-(safe|optimized)$/.test(scenario)) {
+    // convert embroider-xy to embroiderXy
+    const embroiderScenario = scenario
+      .split("-")
+      .map((part, i) =>
+        i === 0 ? part : part.charAt(0).toUpperCase() + part.slice(1)
+      )
+      .join("");
```

Comment:
Could we use a library function instead?

Reply:
As discussed, we can't use `camelize` from `@ember/string` since we can't use ember modules in node.js code. Pulling in an extra library for this small use case would be overkill.